src/main/java/org/terasology/launcher/packages/JenkinsStorage.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.util.JobResult;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Provides game packages from our Jenkins server.
+ */
+public class JenkinsStorage implements Storage {
+
+    private static final Logger logger = LoggerFactory.getLogger(JenkinsStorage.class);
+
+    private static final String JENKINS_JOB_URL = "http://jenkins.terasology.org/job/";
```

Comment:
Much like the game package definition I would load these things from config files as well, not code (might already be intended for a future version?)